bundle/regal/rules/style/rule_name_repeats_package.rego

```diff
@@ -23,51 +34,41 @@ titleize(str) := result if {
 	)
 }
 
-package_path := input["package"].path
-
-package_path_components := [component.value |
-	some component in package_path
+_package_path_components := [component.value |
+	some component in input["package"].path
 	component.type == "string"
 ]
 
-num_package_path_components := count(package_path_components)
+_num_package_path_components := count(_package_path_components)
 
-possible_path_component_combinations contains combination if {
-	some end in numbers.range(1, num_package_path_components)
+_possible_path_component_combinations contains combination if {
+	some end in numbers.range(1, _num_package_path_components)
 
 	combination := array.slice(
-		package_path_components,
-		num_package_path_components - end,
-		num_package_path_components,
+		_package_path_components,
+		_num_package_path_components - end,
+		_num_package_path_components,
 	)
 }
 
-possible_offending_prefixes contains prefix if {
-	some combination in possible_path_component_combinations
+_possible_offending_prefixes contains prefix if {
+	some combination in _possible_path_component_combinations
 
 	prefix := concat("_", combination)
```

Comment:
Should we also catch things like this with this rule?

```rego
package foo.bar

import rego.v1

foo.bar := true
```

Reply:
Good question! I'd say no, as at least applying the same logic as we currently do for _any_ path component would mean that something like:

```rego
package region.us.policies

user.permissions := { }
```

Would count `user` as repeating `us`... 

I think even for a simple naming scheme, this can sometimes be problematic. Like this one for example:

```rego
package my.db

dba if {
    "dba" in input.user.roles
}
```

A config toggle to only match full words would be nice. But it's also not a problem I've enountered in any real policy yet.